deepcell/data/tracking.py

```diff
@@ -0,0 +1,174 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Dataset Builders"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import math
+
+import numpy as np
+import tensorflow as tf
+import tensorflow_addons as tfa
+
+from deepcell.data import split_dataset
+
+
+def temporal_slice(X, y, track_length=8):
+    """Randomly slice movies and labels with a length of ``track_length``.
+
+    Args:
+        X (dict): Dictionary of feature data.
+        y (dict): Dictionary of labels.
+        track_length (int): Length of temporal slices.
+
+    Returns:
+        tuple(dict, dict): Tuple of sliced ``X`` and ``y`` data.
+    """
+    # TODO: time axis may change! would be easier if it was always first.
+    appearances = X['appearances']
+    max_time = tf.shape(appearances)[1] - track_length
+
+    t_start = tf.random.uniform(shape=[], minval=0,
+                                maxval=max_time,
+                                dtype=tf.int32)
+
+    t_end = t_start + track_length
+
+    for key, data in X.items():
+        X[key] = data[t_start:t_end]
+
+    for key, data in y.items():
+        y[key] = data[t_start:t_end - 1]
+
+    return (X, y)
+
+
+def random_rotate(X, y, rotation_range=0):
+    """Randomly rotate centroids and appearances.
+
+    Args:
+        X (dict): Dictionary of feature data.
+        rotation_range (int): Maximum rotation range in degrees.
+
+    Returns:
+        dict: Rotated ``X`` data.
+    """
+    appearances = X['appearances']
+    centroids = X['centroids']
+
+    # Calculate the random rotation in radians
+    rg = rotation_range * math.pi / 180
+    theta = tf.random.uniform(shape=[1], minval=-rg, maxval=rg)
+
+    # Transform appearances
+    # TODO: what is being reshaped here? How to convert to time-first?
+    old_shape = tf.shape(appearances)
+    new_shape = [-1, old_shape[2], old_shape[3], old_shape[4]]
+    img = tf.reshape(appearances, new_shape)
+    img = tfa.image.rotate(img, theta)
+    img = tf.reshape(img, old_shape)
+    X['appearances'] = img
+
+    # Rotate coordinates
+    cos_theta = tf.math.cos(theta)
+    sin_theta = tf.math.sin(theta)
+    rotation_matrix = tf.concat([
+        [cos_theta, -sin_theta],
+        [sin_theta, cos_theta],
+    ], axis=1)
+    transformed_centroids = tf.matmul(centroids, rotation_matrix)
+    X['centroids'] = transformed_centroids
+
+    return X, y
+
+
+def random_translate(X, y, range=512):
+    """Randomly translate the centroids."""
+    centroids = X['centroids']
+    r0 = tf.random.uniform([1, 1, 2], -range, range)
```

Comment:
This is a good question. I would like to handle this in the `graph_tracking` branch though, since this PR does not change the actual augmentation functions, just the way that they are used (`prepare_dataset` vs `DatasetBuilder`).